File: python/aws-lambda/security/tainted-html-string.py
def lambda_handler(event, context):
  # ruleid: tainted-html-string
	foobar(f"<div>{event['input']}</div>")

	response = {}

	link1 = '<a href="http://external/abc/%s">Check link href</a>'
  # ruleid: tainted-html-string
	response['html1'] = link1 % event['input']

  # ruleid: tainted-html-string
	response['html2'] = '<a href="http://external/abc/' + event['input'] + '">Check link href</a>'


	link2 = '<a href="http://external/abc/{}">Check link href</a>'
  # ruleid: tainted-html-string
	response['html3'] = link2.format(event['input'])

	print(f"<div>{event['input']}</div>")

	link_ok = '<a href="http://external/abc/{}">Check link href</a>'
	response['html3'] = link_ok.format("123")

	response['message'] = 'Data: [' + event['input'] + ']'

	result = {
		"statusCode": 200,
		"body": response
	}
	return result

File: javascript/aws-lambda/security/tainted-html-string.js
const {format} = require('util');

exports.handler = async function (event, context) {
  // ruleid: tainted-html-string
  await sendThisHtmlSomewhere(`<h1>message: ${event.name}</h1>`)
  
  // ruleid: tainted-html-string
  const htmlResult = "<h1>" + "message: " + event['message'] + "</h1>";

  let html = "<h1> message"
  // ruleid: tainted-html-string
  html = html.concat(event.message)
  html = html.concat("</h1>")
  doSmth(html)

  // ruleid: tainted-html-string
  foobar(format('<div>Message: %s</div>', event.body.name))

  // ok: tainted-html-string
  foobar(format('Message: %s', event.body.name))

  // ok: tainted-html-string
  console.log('<div>Message: %s</div>', event.body.name)
  
  // ok: tainted-html-string
  console.error(`<h1>message: ${event.name}</h1>`)

  return { body: htmlResult }
}
